lib/logstash/filters/mutate.rb

```diff
@@ -288,6 +291,17 @@ def convert_float(value)
     return value.to_f
   end # def convert_float
 
+  def convert_boolean(value)
+    if value == "true"
+      return true
+    elsif value == "false"
+      return false
+    else
+      @logger.warn("Failed to convert #{value} into boolean. Acceptable values are \"true\" and \"false\".")
+      return value
```

Comment:
@jsvd and @purbon 

`next` doesn't seem to be doing what we expect.


### Updated convert method:

```
  def convert_boolean(value)
    return true if value =~ (/^(true|t|yes|y|1)$/i)
    return false if value.empty? || value =~ (/^(false|f|no|n|0)$/i)
    @logger.warn("Failed to convert #{value} into boolean.")
    nil
  end # def convert_boolean
```
### And I added a check for a `nil`:

```
  def convert(event)
    @convert.each do |field, type|
      next unless event.include?(field)
      original = event[field]
      # calls convert_{string,integer,float,boolean} depending on type requested.
      converter = method("convert_" + type)
      if original.nil?
        next
      elsif original.is_a?(Hash)
        @logger.debug("I don't know how to type convert a hash, skipping",
                      :field => field, :value => original)
        next
      elsif original.is_a?(Array)
        value = original.map { |v| converter.call(v) }
      else
        value = converter.call(original)
      end
      if value.nil?
        next
      else
        event[field] = value
      end
    end
  end # def convert
```

### The test:

```
  describe "add_field fails on boolean conversion failure" do
    config <<-CONFIG
      filter {
        mutate {
          convert   => { "not_bool" => "boolean" }
          add_field => { "potatoes" => "1" }
        }
      }
    CONFIG
    event = {
      "not_bool" => "none of the above"
    }
    sample event do
      expect(subject["not_bool"]).to eq("none of the above")
      expect(subject["potatoes"]).not_to exist
    end
  end
```

### The test fails

Clearly `event["potatoes"]` got the value "1":

```
Failures:

  1) LogStash::Filters::Mutate add_field fails on boolean conversion failure "{"not_bool":"none of the above"}" when processed
     Failure/Error: Unable to find matching line from backtrace
     NoMethodError:
       "1" does not respond to either #exist? or #exists?
     # ./spec/filters/mutate_spec.rb:177:in `(root)'
```